### PR TITLE
feat: update-delete-routes

### PR DESCRIPTION
**Pull request by mosidrum**

### CODE REVIEW

File: src/router/shortUrl.ts [hunk 1]
```diff
@@ -13,3 +19,11 @@ router.post("/urls", createUrlValidationRule, validateRequest, postUrl);
 router.get("/urls/:id", getUrlValidationRule, validateRequest, getOneUrl);
 
 router.get("/urls", validateRequest, getAllUrl);
+router.put(
+  "/urls/:id",
+  getUrlValidationRule,
+  createUrlValidationRule,
```
Comment (greatertomi) [inline src/router/shortUrl.ts]: For `PUT`, all data should be optional according to the spec. This validation is going to enforce you send data

File: src/router/shortUrl.ts [hunk 2]
```diff
@@ -13,3 +19,11 @@ router.post("/urls", createUrlValidationRule, validateRequest, postUrl);
...
+  validateRequest,
+  updateOne,
+);
+router.delete("/urls/:id", getUrlValidationRule, validateRequest, deleteOne);
```
Comment (greatertomi) [inline src/router/shortUrl.ts]: Give `getUrlValidationRule` a more generic name that would make it work for both get and delete route

File: src/controller/deleteOne.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        res,
+        statusCode: STATUS_CODES.NOT_FOUND,
+        message: "URL not found",
+      });
+    }
+
+    return sendResponse({
+      res,
+      statusCode: STATUS_CODES.OK,
+      message: "Deleted succesfully",
```
Comment (greatertomi) [inline src/controller/deleteOne.ts]:
```suggestion
      message: "Deleted successfully",
```

**Review by greatertomi — CHANGES_REQUESTED** [review]

File: src/router/shortUrl.ts (hunk 1 above)
Comment (mosidrum) [reply src/router/shortUrl.ts]: I have made the originalUrl optional.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (greatertomi) [inline .gitignore]: Why this empty space?
```suggestion
```

File: src/router/shortUrl.ts [hunk 5]
```diff
@@ -1,15 +1,30 @@
...
 
 router.get("/urls", validateRequest, getAllUrl);
+router.put(
+  "/urls/:id",
+  validateParamsId,
+  updateOneValidationRule,
+  validateRequest,
+  updateOne,
+);
+router.delete("/urls/:id", validateParamsId, validateRequest, deleteOne);
```
Comment (greatertomi) [inline src/router/shortUrl.ts]:
```suggestion
router.delete("/urls/:id", idParamRule, validateRequest, deleteOne);
```

**Review by greatertomi — CHANGES_REQUESTED** [review]

**Review by greatertomi — APPROVED** [review]

**Pull request merged**